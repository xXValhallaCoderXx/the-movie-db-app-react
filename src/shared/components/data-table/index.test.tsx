import React from "react";
import {shallow} from "enzyme";
import DataTable from "./view-table";

describe("Data Table Component", () => {
  const cols = [
    {header: "ID", name: "id"},
    {header: "Name", name: "name"},
    {header: "Email", name: "email"}
  ];
  const data = [
    {id: 5, name: "John", email: "john@example.com"},
    {id: 6, name: "Liam", email: "liam@example.com"},
    {id: 7, name: "Maya", email: "maya@example.com", someTest: 10},
    {id: 8, name: "Oliver", email: "oliver@example.com", hello: "hello world"},
    {id: 25, name: "Amelia", email: "amelia@example.com"}
  ];
  const onClick = jest.fn();
  it("should render correctly", () => {
    const component = shallow(
      <DataTable loading={false} cols={cols} data={data} onClick={onClick} />
    );
    expect(component).toMatchSnapshot();
  });
  it("should render rows correctly with data", () => {
    const component = shallow(
      <DataTable loading={false} cols={cols} data={data} onClick={onClick} />
    );
    // const firstTableRow = component.at(1).find('table').find('tbody').find('tr').first();

    const table = component.find("table");
    expect(table).toHaveLength(1);

    // const thead = table.find("thead");
    // expect(thead).toHaveLength(1);

    // const headers = thead.find("th");
    // expect(headers).toHaveLength(Object.keys(data[0]).length);

    // headers.forEach((th, idx) => {
    //   expect(th.text()).toEqual(data[idx].id);
    // });
    // const tbody = table.find("tbody");
    // expect(tbody).toHaveLength(1);

    // const rows = tbody.find("tr");
    // expect(rows).toHaveLength(data.length);

    // rows.forEach((tr, rowIndex) => {
    //   const cells = tr.find("td");
    //   expect(cells).toHaveLength(data.length);
    //   expect(cells.at(0).text()).toEqual(data[rowIndex].id);
    // });
  });

  // it("should render if no data", () => {
  //   const cols2 = [
  //     {header: "ID", name: "id"},
  //     {header: "Name", name: "name"},
  //     {header: "Email", name: "email"}
  //   ];
  //   const component = mount(
  //     <DataTable cols={cols2} data={[]} onClick={onClick} />
  //   );

  //   const table = component.find("table");
  //   expect(table).toHaveLength(1);

  //   const thead = table.find("thead");
  //   expect(thead).toHaveLength(1);

  //   const headers = thead.find("th");
  //   expect(headers).toHaveLength(cols.length);

  //   headers.forEach((th, idx) => {
  //     expect(th.text()).toEqual(cols[idx].header);
  //   });

  //   const tbody = table.find("tbody");
  //   expect(tbody).toHaveLength(1);

  //   const row = tbody.find("tr");
  //   expect(row).toHaveLength(1);

  //   const cell = row.find("td");
  //   expect(cell).toHaveLength(1);

  //   expect(cell.prop("colSpan")).toEqual(cols.length);

  //   expect(cell.text()).toEqual("There is no data in this table");
  // });
});
