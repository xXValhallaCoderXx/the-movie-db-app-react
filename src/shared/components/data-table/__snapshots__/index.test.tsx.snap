// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Component should render correctly 1`] = `
<table
  className="text-sm w-full bg-white shadow-md rounded my-6"
>
  <thead>
    <tr>
      <th
        className="py-4 px-6 bg-grey-lightest font-bold uppercase text-sm text-grey-dark border-b border-grey-light text-left"
        key="id"
      >
        ID
      </th>
      <th
        className="py-4 px-6 bg-grey-lightest font-bold uppercase text-sm text-grey-dark border-b border-grey-light text-left"
        key="name"
      >
        Name
      </th>
      <th
        className="py-4 px-6 bg-grey-lightest font-bold uppercase text-sm text-grey-dark border-b border-grey-light text-left"
        key="email"
      >
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="5"
    >
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="id"
      >
        5
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="name"
      >
        John
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="email"
      >
        john@example.com
      </td>
    </tr>
    <tr
      key="6"
    >
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="id"
      >
        6
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="name"
      >
        Liam
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="email"
      >
        liam@example.com
      </td>
    </tr>
    <tr
      key="7"
    >
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="id"
      >
        7
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="name"
      >
        Maya
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="email"
      >
        maya@example.com
      </td>
    </tr>
    <tr
      key="8"
    >
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="id"
      >
        8
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="name"
      >
        Oliver
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="email"
      >
        oliver@example.com
      </td>
    </tr>
    <tr
      key="25"
    >
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="id"
      >
        25
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="name"
      >
        Amelia
      </td>
      <td
        className="py-4 px-6 border-b border-grey-light cursor-pointer"
        key="email"
      >
        amelia@example.com
      </td>
    </tr>
  </tbody>
</table>
`;
